apps/ex_nvr_web/lib/ex_nvr_web/controllers/api/event_controller.ex

```diff
@@ -0,0 +1,79 @@
+defmodule ExNVRWeb.API.EventController do
+  use ExNVRWeb, :controller
+
+  action_fallback ExNVRWeb.API.FallbackController
+  plug(:extract_device when action in [:create, :index])
+  plug(:ensure_type when action in [:create, :index])
+
+  alias Plug.Conn
+  alias ExNVR.{Events, Devices}
+
+  @spec create(Conn.t(), map()) :: {:error, Ecto.Changeset.t()} | Conn.t()
+  def create(%Conn{body_params: params} = conn, _) do
+    device = conn.assigns.device
+    type = conn.assigns.type
+
+    with {:ok, event} <- Events.create(params, device, type),
+         event <- format_event(event, device) do
+      conn
+      |> put_status(201)
+      |> json(%{event: event})
```

Comment:
Let's simplify the endpoints for now:

- Create endpoint:
You can just write the endpoint as follows:
```elixir
device_id = Map.get(query_params, "device_id", "")
event_type = Map.get(query_params, "event_type", "")

with %Device{} = device <- Devices.get(device_id),
        true <- event_type == "lpr",
        {:ok, params, plate_picture, full_picture} <- format_params(device, body_params),
        {:ok, _event} <- Events.create_lpr_event(device, params, plate_picture, full_picture) do
   send_resp(conn, 201, "")
end
```

This is a webhook, so no need to send the actual created event.
In `format_event`, check the vendor of the camera and then format the event per that vendor specs.